packages/thirdweb/src/transaction/transaction-store.ts

```diff
@@ -0,0 +1,106 @@
+import type { Chain } from "../chains/types.js";
+import type { ThirdwebClient } from "../client/client.js";
+import type { Hex } from "../utils/encoding/hex.js";
+import { stringify } from "../utils/json.js";
+import { LocalStorage } from "../wallets/in-app/web/utils/Storage/LocalStorage.js";
+import type { Account } from "../wallets/interfaces/wallet.js";
+import { waitForReceipt } from "./actions/wait-for-tx-receipt.js";
+import type { TransactionReceipt } from "./types.js";
+
+type StoredTransaction = {
+  transactionHash: Hex;
+  status: "PENDING" | "CONFIRMED";
+  receipt?: TransactionReceipt;
+};
+
+let storage: LocalStorage | undefined;
+
+/**
+ * Retrieve transactions for the given account for the current session. Sessions are stored in local storage when possible, with memory as a fallback.
+ * @param account {Account} - The account to retrieve transactions for.
+ * @returns {StoredTransaction[]} An array of stored pending and confirmed transactions.
+ */
+export async function getSessionTransactions(
+  account: Account,
+): Promise<StoredTransaction[]> {
+  // No transactions if the store hasn't been initialized yet.
+  if (!storage) {
+    return [];
+  }
+
+  const transactionStore = await getTransactionsFromStorage(storage);
+  return transactionStore[account.address] ?? [];
+}
+
+/**
+ * @internal
+ */
+export async function addTransactionToSession(options: {
+  account: Account;
+  transactionHash: Hex;
+  client: ThirdwebClient;
+  chain: Chain;
+}) {
+  const { account, transactionHash, client, chain } = options;
+  storage =
+    storage ??
+    new LocalStorage({
+      clientId: client.clientId,
+    });
+  // Definitely typed storage
+  const _storage = storage;
+
+  const transactionStore = await getTransactionsFromStorage(_storage);
+  const transactionsForAccount = transactionStore[account.address];
+  transactionStore[account.address] = [
+    ...(transactionsForAccount ?? []),
+    { transactionHash: transactionHash, status: "PENDING" },
+  ];
+  await setTransactionsInStorage(_storage, transactionStore);
+
+  waitForReceipt({
```

Comment:
idea:
1. instead of doing this here simply _only_ store the tx hash
2. in the _list_ for each tx hash (from this list) retrieve the tx information (can probably use waitForReceipt, it _should?_ instantly resolve for tx that are already mined and ready?)

this way you only do this extra work when the user actually cares (they are looking at the tx) at the cost of having to show some loading state while u resolve (seems OK)

Reply:
this also means you can simply "infer" the `state` of a tx, if u have a receipt it's mined, if not it's pending